src/test/scala/nl.knaw.dans.easy.deposit/docs/DatasetXmlSpec.scala

```diff
@@ -22,22 +22,73 @@ import better.files.File
 import javax.xml.XMLConstants
 import javax.xml.transform.Source
 import javax.xml.transform.stream.StreamSource
-import javax.xml.validation.SchemaFactory
+import javax.xml.validation.{ Schema, SchemaFactory }
 import nl.knaw.dans.easy.deposit.TestSupportFixture
 import nl.knaw.dans.easy.deposit.docs.DatasetMetadata.{ DateQualifier, QualifiedSchemedValue }
 import resource.Using
 
 import scala.util.{ Failure, Success, Try }
-import scala.xml.{ Node, Utility }
+import scala.xml.{ Elem, Node, PrettyPrinter, Utility }
 
-class DatasetXmlSpec extends TestSupportFixture {
+trait DdmBehavior {
+  this: TestSupportFixture =>
 
-  private lazy val triedSchema = Try { // loading postponed until we actually start validating
+  lazy val triedSchema: Try[Schema] = Try { // loading postponed until we actually start validating
     val factory = SchemaFactory.newInstance(XMLConstants.W3C_XML_SCHEMA_NS_URI)
     val schemas = "http://dublincore.org/schemas/xmls/qdc/2008/02/11/dcterms.xsd" ::
       "https://easy.dans.knaw.nl/schemas/md/ddm/ddm.xsd" :: Nil
     factory.newSchema(schemas.map(xsd => new StreamSource(xsd)).toArray[Source])
   }
+  val emptyDDM: Elem
+
+  // pretty provides friendlier trouble shooting for complex XML's than Utility.trim
+  private val prettyPrinter: PrettyPrinter = new scala.xml.PrettyPrinter(1024, 2)
+
+  /**
+   * @param input          a Success(DatasetMetadata)
+   * @param expectedOutput <ddm:profile> and optionaly <ddm:dcmiMetadata>
+   *                       if provided, input should convert into a DDM with the same content
+   */
+  def validDatasetMetadata(input: Try[DatasetMetadata], expectedOutput: Seq[Node]): Unit = {
+    lazy val datasetMetadata = input.recoverWith { case e =>
+      println(s"$e")
+      Failure(e)
+    }.getOrElse(fail("test input should be a success"))
+    lazy val ddm = DatasetXml(datasetMetadata).recoverWith { case e =>
+      println(s"$e")
+      Failure(e)
+    }.getOrElse(fail("can't create DDM from test input"))
+
+    it should "generate expected DDM" in {
+      // TODO change into "should contain": leafs in expected should occur with same path in ddm
+      // perhaps with something similar to https://stackoverflow.com/questions/39001421/
+      // or a third argument (with xpaths?) to extract subsets from the generated DDM
+      val expected = emptyDDM.copy(child = expectedOutput)
+      if(expectedOutput.nonEmpty)
+        prettyPrinter.format(ddm) shouldBe
```

Comment:
in [easy-split-multi-deposit](https://github.com/rvanheest-DANS-KNAW/easy-split-multi-deposit/blob/ea7c2dc3d6a78cacd779d2efcb2d799e399e22cf/src/test/scala/nl.knaw.dans.easy.multideposit/CustomMatchers.scala#L41) I made a `should ...` operator for comparing XMLs. Maybe you can use that one here as well?

Reply:
I would not like to loose the IntelliJ supported diff, nor have to wade through big one liners for trouble shooting.

Reply:
https://github.com/DANS-KNAW/easy-split-multi-deposit/pull/113

I'll also copy this to easy-bag-index